ts/torch_handler/vision_handler.py

```diff
@@ -19,44 +17,12 @@ class VisionHandler(BaseHandler, ABC):
     Base class for all vision handlers
     """
 
-    def initialize(self, context):
-        super().initialize(context)
-        self.ig = IntegratedGradients(self.model)
-        self.initialized = True
-        properties = context.system_properties
-        if not properties.get("limit_max_image_pixels"):
-            Image.MAX_IMAGE_PIXELS = None
-
-    def preprocess(self, data):
-        """The preprocess function of MNIST program converts the input data to a float tensor
-
-        Args:
-            data (List): Input data from the request is in the form of a Tensor
-
-        Returns:
-            list : The preprocess function returns the input image as a list of float tensors.
-        """
-        images = []
-
-        for row in data:
-            # Compat layer: normally the envelope should just return the data
-            # directly, but older versions of Torchserve didn't have envelope.
-            image = row.get("data") or row.get("body")
-            if isinstance(image, str):
-                # if the image is a string of bytesarray.
-                image = base64.b64decode(image)
-
-            # If the image is sent as bytesarray
-            if isinstance(image, (bytearray, bytes)):
-                image = Image.open(io.BytesIO(image))
-                image = self.image_processing(image)
-            else:
-                # if the image is a list
-                image = torch.FloatTensor(image)
-
-            images.append(image)
-
-        return torch.stack(images).to(self.device)
+    image_processing = transforms.Compose([transforms.ToTensor()])
+
+    def __init__(self):
+        super().__init__()
+        self.initialize = VisionInitialize(BaseInitialize())
```

Comment:
mmm I don't quite like this as much as the rest, to me it feels like i should call one initialize or the other

Which reminds me it's kinda funky that we have both init and initialize

Reply:
Agreed, for initialization its kind of weird. This reflects the previous behavior. We are calling [super().initialize() in VisionHandler](https://github.com/pytorch/serve/blob/6daaa428c855c6eff6831807bba46f109b623d79/ts/torch_handler/vision_handler.py#L23). I think if you break up the handler method like the pre- and postprocessing it could make sense.
```python
class CustomHandler(BaseHandler):
   def __init__(self):
      init = LoadModel()
      init = LoadWeights(init)
      init = Quantize(init, np.int8, layer_names)
      init = Compile(init, method='max-autotune')
      self.initialize = init
```